bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -2917,7 +2917,9 @@ ReplicaImp::ReplicaImp(bool firstTime,
   }
 
   std::set<NodeIdType> clientsSet;
-  for (uint16_t i = config_.numReplicas; i < config_.numReplicas + config_.numOfClientProxies; i++)
+  for (uint16_t i = config_.numReplicas;
+       i < config_.numReplicas + config_.numOfClientProxies + config_.numOfExternalClients;
```

Comment:
Don't re-calculate the loop end condition at every iteration: define a const variable before a loop and use it inside: 
const auto numOfEntities = config_.numReplicas + config_.numOfClientProxies + config_.numOfExternalClients;
for (uint16_t i = config_.numReplicas; I < numOfEntities....